File: python_leaoai/python_leaoai/services/ApplicationServices/Core/PageSectionService.py
from typing import List, Optional

class PageSectionService:
    def __init__(self, page_section_repository):
        self._page_section_repository = page_section_repository

    def get_page_sections_by_owner(self, owner_id: int, owner_type: str) -> List:
        """Retorna todas as PageSections de um dono específico"""
        return self._page_section_repository.get_by_owner(owner_id, owner_type)

    def get_by_id(self, id: int):
        """Retorna uma PageSection pelo ID"""
        return self._page_section_repository.get_by_id(id)

    def add_page_section(self, page_section):
        """Adiciona uma nova PageSection"""
        self._page_section_repository.add(page_section)

    def update_page_section(self, page_section):
        """Atualiza uma PageSection existente"""
        self._page_section_repository.update(page_section)

    def delete_page_section(self, page_section_id: int):
        """Marca uma PageSection como excluída e inativa"""
        page_section = self._page_section_repository.get_by_id(page_section_id)
        if page_section:
            page_section.excluido = True
            page_section.ativo = False
            self._page_section_repository.update(page_section)

File: Framework.Services/ApplicationServices/Core/PageSectionService.cs
using Framework.Data.Models.Conteudo;
using Framework.Repositories.Interface.Core;
using Framework.Services.Interfaces.Core;
using System.Collections.Generic;

namespace Framework.Services.ApplicationServices.Core
{
    public class PageSectionService : IPageSectionService
    {
        private readonly IPageSectionRepository _pageSectionRepository;

        public PageSectionService(IPageSectionRepository pageSectionRepository)
        {
            _pageSectionRepository = pageSectionRepository;
        }

        // Métodos para PageSection

        public IEnumerable<PageSection> GetPageSectionsByOwner(int ownerId, string ownerType)
        {
            return _pageSectionRepository.GetPageSectionsByOwner(ownerId, ownerType);
        }

        public PageSection GetById(int id)
        {
            return _pageSectionRepository.GetById(id);
        }

        public void AddPageSection(PageSection pageSection)
        {
            _pageSectionRepository.Add(pageSection);
        }

        public void UpdatePageSection(PageSection pageSection)
        {
            _pageSectionRepository.Update(pageSection);
        }

        public void DeletePageSection(int pageSectionId)
        {
            PageSection pageSection = _pageSectionRepository.GetById(pageSectionId);
            if (pageSection != null)
            {
                pageSection.Excluido = true;
                pageSection.Ativo = false;

                _pageSectionRepository.Update(pageSection);
            }
        }
    }

}
